screens/modals/RewardModal.js

```diff
@@ -0,0 +1,185 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import { Modal, View } from 'react-native';
+import Colors from '../../assets/Colors';
+import { ButtonLabel, Title } from '../../components/BaseComponents';
+import { RewardAppliedContainer, RewardAvailableContainer } from '../../styled/checkout';
+import { ModalCenteredOpacityLayer } from '../../styled/modal';
+import { ColumnContainer, RoundedButtonContainer, RowContainer } from '../../styled/shared';
+
+export default class RewardModal extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      modalVisible: false,
+      rewardsApplied: 0,
+      rewardStatus: [],
+      totalPrice: 0,
+      isLoading: true
+    };
+  }
+
+  componentDidMount() {
+    const { rewardsAvailable, rewardsApplied, totalPrice } = this.props;
+    const rewardStatus = [];
+    // Build initial status array
+    for (let i = 0; i < rewardsAvailable; i += 1) {
+      if (i < rewardsApplied) {
+        rewardStatus.push(true);
+      } else {
+        rewardStatus.push(false);
+      }
+    }
+    this.setState({
+      rewardsApplied,
+      rewardStatus,
+      totalPrice,
+      isLoading: false
+    });
+  }
+
+  componentWillReceiveProps({ totalPrice }) {
+    this.setState(prevState => ({ ...prevState, totalPrice }));
+  }
+
+  setModalVisible = visible => this.setState({ modalVisible: visible });
+
+  handleShowModal = () => {
+    const { totalPrice, rewardsAvailable, rewardsApplied } = this.props;
+    if ((totalPrice < 5 && rewardsApplied === 0) || rewardsAvailable === 0) {
+      return;
+    }
+    this.setModalVisible(!this.state.modalVisible);
+  };
+
+  handleApplyRewards = () => {
+    // Communicate to parent component
+    this.props.callback(this.state.rewardsApplied, this.state.totalPrice);
+    this.setModalVisible(!this.state.modalVisible);
```

Comment:
Same comment as earlier modal, is toggle actually necessary or would `this.setModalVisible(false)` suffice